basyx.aasenvironment/basyx.aasenvironment-client/src/test/java/org/eclipse/digitaltwin/basyx/aasenvironment/client/TestFixture.java

```diff
@@ -91,6 +94,15 @@ public AssetAdministrationShellDescriptor buildAasPre1Descriptor() {
 		return DummyAasDescriptorFactory.createDummyDescriptor(AAS_PRE1_ID, AAS_PRE1_IDSHORT, AAS_PRE1_GLOBALASSETID, aasRepositoryBasePath);
 	}
 
+	public AssetAdministrationShellDescriptor buildAasPre1Descriptor_withMultipleInterfaces() {
+		LinkedList<Endpoint> endpoints = new LinkedList<>();
+
+		endpoints.add(DummyAasDescriptorFactory.createEndpoint(AAS_POS1_ID, aasRepositoryBasePath, "OTHER-3.0"));
```

Comment:
I would change this to `AAS-REPOSITORY-3.0` and create an endpoint with just the path to the AAS Repository because `OTHER-3.0` would not be spec conform even if this is just for testing purposes.

Reply:
The problem is that we do not test whether it chooses the correct interface, since both endpoints have the same one. 

The current implementation takes into account
1. Is the interface correct?
2. Is the endpoint reachable? (via a simple HTTP HEAD request)

If we have two endpoints with AAS-REPOSITORY-3.0, it'll take the first one. If the first is a simple path to the AAS repository, then both checks will succeed; which will result in it wrongly being selected for AAS resolution.